src/routes/safe/store/models/transaction.ts

```diff
@@ -1,42 +1,106 @@
-import { List, Record } from 'immutable'
+import { List, Map, Record } from 'immutable'
 
 import { ZERO_ADDRESS } from 'src/logic/wallets/ethAddresses'
 
-export const OUTGOING_TX_TYPE = 'outgoing'
+export type TransactionType =
+  | 'incoming'
+  | 'outgoing'
+  | 'settings'
+  | 'custom'
+  | 'creation'
+  | 'cancellation'
+  | 'upgrade'
+  | 'token'
+  | 'collectible'
+
+export type TransactionStatus =
+  | 'awaiting_your_confirmation'
+  | 'awaiting_confirmations'
+  | 'success'
+  | 'failed'
+  | 'cancelled'
+  | 'awaiting_execution'
+  | 'pending'
+
+export type PendingActionType = 'Confirm' | 'Reject'
+
+export type TransactionProps = {
+  baseGas: number
+  blockNumber?: number | null
+  cancelled?: boolean
+  confirmations: List<any>
+  creationTx: boolean
+  customTx: boolean
+  data?: string | null
+  decimals?: (number | string) | null
+  decodedParams: any
+  executionDate?: string | null
+  executionTxHash?: string | null
+  executor: string
+  gasPrice: number
+  gasToken: string
+  isCancellationTx: boolean
+  isCollectibleTransfer: boolean
+  isExecuted: boolean
+  isPending?: boolean
+  isSuccessful: boolean
+  isTokenTransfer: boolean
+  modifySettingsTx: boolean
+  multiSendTx: boolean
+  nonce?: number | null
+  operation: number
+  origin: string | null
+  ownersWithPendingActions: Map<PendingActionType, List<any>>
```

Comment:
is there actually a possibility for us to know pending actions for all the owners? or this is actually done with mind that user could have multiple wallets?

Reply:
Exactly that. I was in the direction of a simple 'pending' status but ended up picking the work done in #778

Reply:
I've thought about a different approach and I will tell you about it in a separate comment